### PR TITLE
SCW integration

### PR DESCRIPTION
**Pull request by renanmav**
# Smart Contract Wallet (SCW) integration

This PR integrates SCW into our app. It uses passkey authentication and viem.

- https://webauthn.guide/#intro
- https://viem.sh/account-abstraction

## Demos

| iOS | Android |
| --- | --- |
| <video src="https://github.com/user-attachments/assets/bd44f7a2-f244-4887-a88e-3888b9276d86" /> | TODO |

## Checklist and follow-ups

- [x] Works on iOS
- [ ] Add viem setup
- [ ] Add API calls
- [ ] Solve smartAccount serialization problem into redux store
- [ ] Works on Android

### CODE REVIEW

File: ios/liquidinterface/Info.plist [hunk 1]
```diff
@@ -51,9 +53,13 @@
       <key>NSAllowsLocalNetworking</key>
       <true/>
     </dict>
+    <key>NSFaceIDUsageDescription</key>
+    <string>This app uses Face ID to sign in with Passkeys.</string>
     <key>NSUserActivityTypes</key>
     <array>
       <string>$(PRODUCT_BUNDLE_IDENTIFIER).expo.index_route</string>
+      <string>$(PRODUCT_BUNDLE_IDENTIFIER).expo.index_route</string>
+      <string>$(PRODUCT_BUNDLE_IDENTIFIER).expo.index_route</string>
```
Comment (renanmav) [inline ios/liquidinterface/Info.plist:62]: This thing gets added every time I run the prebuild command

File: store/smartAccount/createSmartAccount.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        displayName: username,
+      },
+      pubKeyCredParams: [{ type: 'public-key', alg: -7 }],
+      authenticatorSelection: {
+        authenticatorAttachment: 'platform',
+        userVerification: 'required',
+        residentKey: 'required',
+        requireResidentKey: true,
+      },
+    });
```
Comment (njokuScript) [inline store/smartAccount/createSmartAccount.ts:62]: this will be updated in the pr for API calls right? @renanmav

**Review by njokuScript — APPROVED** [review]

File: store/smartAccount/createSmartAccount.ts (hunk 2 above)
Comment (renanmav) [reply store/smartAccount/createSmartAccount.ts:62]: Yes @njokuScript

Comment (njokuScript) [reply store/smartAccount/createSmartAccount.ts:62]: LGTM!

**Pull request merged**